lib/reek/smells/nested_iterators.rb

```diff
@@ -10,6 +10,16 @@ module Smells
     #
     # See {file:docs/Nested-Iterators.md} for details.
     class NestedIterators < SmellDetector
+      # Struct for conveniently associating iterators with their depth (that is, their nesting).
+      IteratorToDepth = Struct.new :iterator, :depth do
+        include Comparable
+        def <=>(other)
+          depth <=> other.depth
+        end
+      end
```

Comment:
I really dislike the `IteratorToDepth` name. How about just `Iterator`, or `IteratorWithDepth` (bleh…)?